frontend/public/components/monitoring/alert-manager-receiver-forms.tsx

```diff
@@ -0,0 +1,493 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+
+import {
+  ButtonBar,
+  Dropdown,
+  ExternalLink,
+  Firehose,
+  history,
+  LoadingBox,
+  SectionHeading,
+  StatusBox,
+} from '../utils';
+import { getAlertManagerConfig, patchAlertManagerConfig } from './alert-manager-utils';
+import { K8sResourceKind } from '../../module/k8s';
+import { ActionGroup, Button } from '@patternfly/react-core';
+import { Helmet } from 'react-helmet';
+import { RadioInput } from '../radio';
+import { MinusCircleIcon, PlusCircleIcon } from '@patternfly/react-icons';
+import { AlertManagerConfig, AlertManagerReceiver } from './alert-manager-config';
+
+const PagerDutySubForm = ({ pagerDutyValues, setPagerDutyValues }) => {
+  const updatePagerDutyValues = (key, value) => {
+    _.set(pagerDutyValues, key, value);
+    setPagerDutyValues(_.clone(pagerDutyValues));
+  };
+
+  return (
+    <div className="co-m-pane__body--section-heading">
+      <SectionHeading text="PagerDuty Configuration" />
+      <div className="form-group">
+        <label className="control-label">Integration Type</label>
+        <div>
+          <RadioInput
+            title="Events API v2"
+            name="integrationType_Events"
+            id="integration-type-events"
+            value="events"
+            onChange={(e) => updatePagerDutyValues('integrationType', e.target.value)}
+            checked={pagerDutyValues.integrationType === 'events'}
+            inline
+          />
+          <RadioInput
+            title="Prometheus"
+            name="integrationType_Prometheus"
+            id="integration-type-prometheus"
+            value="prometheus"
+            onChange={(e) => updatePagerDutyValues('integrationType', e.target.value)}
+            checked={pagerDutyValues.integrationType === 'prometheus'}
+            inline
+          />
+        </div>
+      </div>
+      <div className="form-group">
+        <label className="control-label co-required">
+          {pagerDutyValues.integrationType === 'events' ? 'Routing' : 'Service'} Key
+        </label>
+        <input
```

Comment:
When testing, I noticed that the `Routing Key` / `Service Key` value is lost if you change the `Receiver Type` to webhook and then change it back.

The Webhook `URL` value is not lost if you change the `Receiver Type` to PagerDuty and back again.

Reply:
yeah, I caught that in the next PR I'm going to post; back ported to this PR

Reply:
Doesn't look like this was back ported to this PR, but OK to handle it in the next PR.